### PR TITLE
Feature rights field

### PR DESCRIPTION
**Pull request by kjgarza**
## Description
Enable rigths addition via doi form

resolves: https://github.com/datacite/bracco/issues/333

## List of General Components affected
- component, test, model and serirliazer
- controller to get rights information.

## Status
- [x] Ready for Review

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Non Functional Requirement
- [x] Follows the code style of this project.
- [ ] Tests Cover Changes
- [x] All new and existing tests passed.
- [ ] Documentation

### CODE REVIEW

File: app/components/doi-funding-reference.js [hunk 1]
```diff
@@ -48,18 +48,28 @@ export default Component.extend({
...
-      this.fragment.set('funderIdentifier', funder.uri);
-      this.set('isCrossrefId', true);
-    } else {
-      this.fragment.set('funderIdentifierType', 'Other');
-      this.fragment.set('funderIdentifier', null);
-      this.fragment.set('funderName', funder);
-      this.updateFunderSchemeAndType(null);
-      this.set('isCrossrefId', false);
+    switch (true) {
+      case funder === null:
```
Comment (richardhallett) [inline app/components/doi-funding-reference.js:52]: I'm not saying this is wrong, but javascript "falsey" can be tricky at times, is it definitely null?

Comment (kjgarza) [reply app/components/doi-funding-reference.js:52]: definetely null.  When the users click on the `X`  in the selector it sends `null` and then we clear all the fields.

File: app/controllers/dois/show/edit.js [hunk 2]
```diff
@@ -17,6 +25,22 @@ export default Controller.extend({
       return 'hide';
     }
   },
+  getSpdxList() {
```
Comment (richardhallett) [inline app/controllers/dois/show/edit.js:28]: Shouldnt this be part of a service?

File: app/controllers/repositories/show/dois/new.js [hunk 3]
```diff
@@ -12,6 +20,22 @@ export default Controller.extend({
       return null;
     }
   },
+  getSpdxList() {
```
Comment (richardhallett) [inline app/controllers/repositories/show/dois/new.js:23]: Again here, so refactor service then remove duplication

**Pull request merged**